pulp_python/tests/functional/api/test_pypi_simple_api.py

```diff
@@ -4,21 +4,83 @@
 import requests
 
 from pulp_python.tests.functional.constants import (
+    PYPI_SERIAL_CONSTANT,
     PYTHON_EGG_FILENAME,
+    PYTHON_EGG_SHA256,
     PYTHON_EGG_URL,
+    PYTHON_SM_FIXTURE_CHECKSUMS,
+    PYTHON_SM_FIXTURE_RELEASES,
     PYTHON_SM_PROJECT_SPECIFIER,
     PYTHON_WHEEL_FILENAME,
+    PYTHON_WHEEL_METADATA_SHA256,
+    PYTHON_WHEEL_SHA256,
     PYTHON_WHEEL_URL,
+    PYTHON_XS_FIXTURE_CHECKSUMS,
 )
+from pulp_python.tests.functional.utils import ensure_simple
 
 API_VERSION = "1.1"
-PYPI_SERIAL_CONSTANT = 1000000000
 
 PYPI_TEXT_HTML = "text/html"
 PYPI_SIMPLE_V1_HTML = "application/vnd.pypi.simple.v1+html"
 PYPI_SIMPLE_V1_JSON = "application/vnd.pypi.simple.v1+json"
 
 
+@pytest.mark.parallel
+def test_simple_html_index_api(
+    python_remote_factory, python_repo_with_sync, python_distribution_factory
+):
+    remote = python_remote_factory(includes=PYTHON_SM_PROJECT_SPECIFIER)
+    repo = python_repo_with_sync(remote)
+    distro = python_distribution_factory(repository=repo)
+
+    url = urljoin(distro.base_url, "simple/")
+    headers = {"Accept": PYPI_SIMPLE_V1_HTML}
+
+    response = requests.get(url, headers=headers)
+    assert response.headers["Content-Type"] == PYPI_SIMPLE_V1_HTML
+    assert response.headers["X-PyPI-Last-Serial"] == str(PYPI_SERIAL_CONSTANT)
+
+    proper, msgs = ensure_simple(
+        url, PYTHON_SM_FIXTURE_RELEASES, sha_digests=PYTHON_SM_FIXTURE_CHECKSUMS
+    )
+    assert proper, f"Simple API validation failed: {msgs}"
+
+
+def test_simple_html_detail_api(
+    delete_orphans_pre,
+    monitor_task,
+    python_bindings,
+    python_content_factory,
+    python_distribution_factory,
+    python_repo_factory,
+):
+    content_1 = python_content_factory(PYTHON_WHEEL_FILENAME, url=PYTHON_WHEEL_URL)
+    content_2 = python_content_factory(PYTHON_EGG_FILENAME, url=PYTHON_EGG_URL)
+    body = {"add_content_units": [content_1.pulp_href, content_2.pulp_href]}
+
+    repo = python_repo_factory()
+    monitor_task(python_bindings.RepositoriesPythonApi.modify(repo.pulp_href, body).task)
+    distro = python_distribution_factory(repository=repo)
+
+    url = f'{urljoin(distro.base_url, "simple/")}shelf-reader'
+    headers = {"Accept": PYPI_SIMPLE_V1_HTML}
+
+    response = requests.get(url, headers=headers)
+    assert response.headers["Content-Type"] == PYPI_SIMPLE_V1_HTML
+    assert response.headers["X-PyPI-Last-Serial"] == str(PYPI_SERIAL_CONSTANT)
+
+    proper, msgs = ensure_simple(
+        urljoin(distro.base_url, "simple/"),
+        {"shelf-reader": [PYTHON_WHEEL_FILENAME, PYTHON_EGG_FILENAME]},
+        sha_digests=PYTHON_XS_FIXTURE_CHECKSUMS,
+    )
+    assert proper, f"Simple API validation failed: {msgs}"
+
+    html_content = response.text
+    assert f'data-dist-info-metadata="sha256={PYTHON_WHEEL_METADATA_SHA256}' in html_content
```

Comment:
Maybe we can add this check into ensure_simple. Pass in a new metadata_sha_digests dict that should check if certain links have this field set correctly.